src/scripts/exclusion_term_expansion/exclusion_term_expansion.py

```diff
@@ -0,0 +1,312 @@
+
+# Resources
+- GitHub issue: https://github.com/monarch-initiative/mondo-ingest/issues/22
+- ICD10CM exclusions, online: https://docs.google.com/spreadsheets/d/1cZPUReTl34Vu2a03tm2921ehARiIpp4fjeQfS27XzKA/edit#gid=1119821904
+- Ophanet/ORDO exclusions, online: https://docs.google.com/spreadsheets/d/16ftBJ8mYqEvSEVNi1tGxIlDfL88vYrs7tSUjSnGqrBw/edit#gid=0
+
+TODO's
+  - todo: Do we need to worry about edge case where exclusion table is empty? (e.g. we include such a table for all
+     ontologies, but there exists 1 where we don't exclude anything.
+  - todo: later: see below: #x1: regex results: would be good to filter out by IRI or prefix that is not from source
+     ontology, but how to determine the appropriate prefix just based on the 2 file params, but could have edge cases.
+  - todo: later: see below: #x2
+  - todo: later: see below: #x3
+  - todo: later: see below: #x4: Switch from SPARQL to OAK. (SqliteImpl() is faster than SparqlWrapper (which uses rdflib)
+  - todo: later: When constructing exclusion TSV: bug: ICD10CM:C7A-C7A reported as term, but should be ICD10CM:C7A
+  - todo: later: QA: possible conflicts in icd10cm_exclusions.tsv: Add a check to raise an error in event of a parent
+     class having `True` for `exclude_children`, but the child class has `False`.
+"""
+import os
+import subprocess
+from argparse import ArgumentParser
+from typing import Dict, List, Union
+
+from jinja2 import Template
+import pandas as pd
+
+
+# Vars
+# # Config
+CONFIG = {
+    'use_cache': False,
+    'include_superclass': False,  # is this the correct name for what this does: * property path qualifier
+    'save': True,  # save dataframes to disk
+    'regex_prefix': 'REGEX:'  # used to know which rows in exclusion TSVs are regex pattern searches
+}
+
+# # Static
+THIS_SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+CACHE_DIR = os.path.join(THIS_SCRIPT_DIR, 'cache')
+SCRIPTS_DIR = os.path.join(THIS_SCRIPT_DIR, '..')
+PROJECT_DIR = os.path.join(SCRIPTS_DIR, '..')
+ONTOLOGY_DIR = os.path.join(PROJECT_DIR, 'ontology')
+CONFIG_DIR = os.path.join(ONTOLOGY_DIR, 'config')
+SPARQL_DIR = os.path.join(PROJECT_DIR, 'sparql')
+SPARQL_CHILD_INCLUSION_PATH = os.path.join(SPARQL_DIR, 'get-terms-children.sparql.jinja2')
+SPARQL_TERM_REGEX_PATH = os.path.join(SPARQL_DIR, 'get-terms-by-regex.sparql.jinja2')
+
+
+# Functions
+def uri_to_curie(uri: str) -> str:
+    """Takes an ontological URI and returns a CURIE. Works on the following patterns:
+    - http://.../PREFIX/CODE
+    - http://.../PREFIX_CODE"""
+    uri_list: List[str] = uri.split('/')
+    end_bit = uri_list[-1]
+    if '_' in end_bit:
+        curie = end_bit.replace('_', ':')
+    else:
+        curie = uri_list[-2] + ':' + end_bit
+    return curie
+
+
+# todo: later #x2: terms vs regexp: this would be better as a class or done some other way probably
+# todo: later #x3: could be nice to add param to auto-do uri_to_curie()
+def sparql_jinja2_robot_query(
```

Comment:
Hmm, I agree this function is not 100% general. It is almost general enough to be given that name, but it does have a couple specifics. I know that the Jinja is out of place a little, but I think it is actually more in line with how we do things in our pipeline (flat files of all different kinds) rather than constructing in runtime. The only thing that'd make it more in line with our pipeline is if I wrote make goals to instantiate the Jinja.